gatorgrade/generate/generate.py

```diff
@@ -1,21 +1,82 @@
-"""Generate a GatorGrade configuration file.
+"""Generate a YAML file with default messages and specific paths."""
+import os
+from typing import List
 
-Gatorgrade file will include paths to whitelisted files
-and default GatorGrader checks.
-"""
+# Define colors for terminal output
+OKGREEN = "\033[92m"
+WARNING = "\033[93m"
+FAIL = "\033[91m"
 
-# Import the necessary libraries
 
-
-def create_targeted_paths_list():
+def create_targeted_paths_list(
+    key_word_list: List[str], relative_run_path: str = "."
+) -> List[str]:
     """Generate a list of targeted paths by walking the paths."""
+    targeted_paths = []
     # Go through the root repo, the sub dictionaries and files.
-    # Select only files in the dictionaries with specific names.
-    # Add those targeted file paths into a list and returns it.
+    # The os.walk will only scan the paths.
+    # So the empty folders containing nothing won't be gone through
+    for dirpath, _, filenames in os.walk(relative_run_path):
+        # Split path string into multiple layers of directories
+        path_dir_list = dirpath.split("/")
+        # Ignore folder starting with double underscore
+        if any(path.startswith("__") for path in path_dir_list):
+            continue
+        # Ignore hidden folders and first layer. the root repo is always dot
+        # Keep double dot. It represents the returning sign
+        if any(
+            path.startswith(".") and not path.startswith("..")
+            for path in path_dir_list[1:]
+        ):
+            continue
+        for filename in filenames:
+            # Ignore the file starting with double underscore and hidden file
+            if filename.startswith("__") or filename.startswith("."):
+                continue
```

Comment:
I don't know how much you've covered, but I see these same checks above and a nested `for` loop, could we use recursion instead or is there a reason we couldn't? 

I'm thinking one reason recursion could be challenging is that below it looks like you are looking a few directories deeper, so you'd have to store state of what you've seen.

Reply:
Also, since these are cases where we skip and we are checking them in a couple places, what do you think about defining a `check_skip_path()` that takes a path string and returns a boolean value of whether or not we should skip.

The reasoning behind this comment is that if we have additional skip cases, we only have to update them in a single place and not search through our codebase.

Reply:
Hi, I can explain some questions above. For the first one, you can see in the first for statement
```   for dirpath, _, filenames in os.walk(relative_run_path):
```
Dirpath and file are treated separately. And filenames is a list of file names. So I was doing the same check, but it was to different subjects.

Reply:
I agree with the function idea. it may make the codes dryer. I may try to do it later. However, I am not so sure about how to achieve recursion here. Could you please leave more details about it?